eclipse-plugin/eclipse/cc.codechecker.eclipse.plugin/src/cc/codechecker/plugin/views/report/details/BugPathListView.java

```diff
@@ -72,8 +72,11 @@ public void setName(String value) {
         setPartName(value);
     }
 
-    public void clear() {
-        viewer.setInput(null);
+    public void clean(boolean delegate) {
+        if(viewer != null && !delegate) {
+            viewer.setInput(null);
+            viewer.refresh();
+        }
```

Comment:
I have introduced this extra variable to make sure the function is called by the good thread. This function is called by a function which is called from two distinct locations. One of the callers are running on the GUI thread which has authority over this element, so it's fine. But the other caller is on a separate thread, so its call effectively crashes because of cross-thread action.